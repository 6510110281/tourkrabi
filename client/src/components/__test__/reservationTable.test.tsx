import { render, screen, fireEvent } from '@testing-library/react';
import DataTable from "../reservationTable";

const reservation = {
  id: 1,
  reservation_date: new Date(),
  reservation_expire_date: new Date(),
  total_price: 100,
  payment_status: false,
  tour: {
    tour_name: "Test Tour",
    current_participate: 0,
    maximun_participate: 10,
  },
};

describe("DataTable", () => {
  test("displays tour name", () => {
    render(<DataTable userReservation={reservation} />);
    expect(screen.getByText(reservation.tour.tour_name)).toBeInTheDocument();
  });

  test("displays current and maximum participants", () => {
    render(<DataTable userReservation={reservation} />);
    expect(screen.getByText(`${reservation.tour.current_participate}/${reservation.tour.maximun_participate}`)).toBeInTheDocument();
  });

  test("displays reservation date", () => {
    render(<DataTable userReservation={reservation} />);
    expect(screen.getByText(reservation.reservation_date.toLocaleDateString())).toBeInTheDocument();
  });

  test("displays total price", () => {
    render(<DataTable userReservation={reservation} />);
    expect(screen.getByText(`${reservation.total_price} บาท`)).toBeInTheDocument();
  });

  test('should match snapshot', () => {
    const { asFragment } = render(<DataTable userReservation={reservation} />);
    expect(asFragment()).toMatchSnapshot();
  });

  test("displays SweetAlert modal when 'ยังไม่ชำระเงิน' button is clicked", () => {
    render(<DataTable userReservation={reservation} />);
    const notPaidButton = screen.getByText("ยังไม่ชำระเงิน");
    fireEvent.click(notPaidButton);
    expect(screen.getByText(`ยอดรวมทั้งสิ้น ${reservation.total_price} บาท`)).toBeInTheDocument();
    expect(screen.getByAltText("Custom image")).toBeInTheDocument();
  });
});