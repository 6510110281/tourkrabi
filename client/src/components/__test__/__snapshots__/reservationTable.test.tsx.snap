// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DataTable should match snapshot 1`] = `
<DocumentFragment>
  Test Tour0/109/3/2566100 บาท
  <button>
    ยังไม่ชำระเงิน
  </button>
</DocumentFragment>
`;
